colosseum/wrappers.py

```diff
@@ -81,3 +84,91 @@ def closing(self, level):
             return self._quotes[level][-1]
         except IndexError:
             raise IndexError('Quotes level out of range')
+
+
+class Shorthand(MutableMapping):
+    """Dictionary-like wrapper to hold shorthand data."""
+    VALID_KEYS = []
+
+    def __init__(self, **kwargs):
+        if self.VALID_KEYS:
+            for key in kwargs:
+                if key not in self.VALID_KEYS:
+                    raise ValueError('Invalid key "{key}". Valid keys are {keys}'.format(key=key,
+                                                                                         keys=self.VALID_KEYS))
+        self._map = kwargs
+
+    def __eq__(self, other):
+        return other.__class__ == self.__class__ and self._map == other._map
+
+    def __setitem__(self, key, value):
+        if self.VALID_KEYS:
+            if key in self.VALID_KEYS:
+                self._map[key] = value
+            else:
+                raise KeyError('Valid keys are: {keys}'.format(keys=self.VALID_KEYS))
+        else:
+            self._map[key] = value
+
+    def __getitem__(self, key):
+        if self.VALID_KEYS:
+            if key in self.VALID_KEYS:
+                return self._map[key]
+        else:
+            return self._map[key]
+
+        raise KeyError('Valid keys are: {keys}'.format(keys=self.VALID_KEYS))
+
+    def __delitem__(self, key):
+        if self.VALID_KEYS:
+            if key in self.VALID_KEYS:
+                self._map.pop(key)
+        else:
+            self._map.pop(key)
+
+    def __len__(self):
+        return len(self._map)
+
+    def __iter__(self):
+        return iter(self.VALID_KEYS) if self.VALID_KEYS else iter(self._map)
+
+    def __repr__(self):
+        map_copy = self._map.copy()
+        items = []
+        for key in self.VALID_KEYS:
+            if key in map_copy:
+                items.append("{key}={value}".format(key=key, value=repr(map_copy[key])))
+
+        class_name = self.__class__.__name__
+        string = "{class_name}({items})".format(class_name=class_name, items=', '.join(items))
+        return string.format(**map_copy)
+
+    def __str__(self):
+        return repr(self)
+
+    def items(self):
+        return sorted(self._map.items())
+
+    def copy(self):
+        return self.__class__(**self._map)
+
+    def to_dict(self):
+        return self._map.copy()
+
+    @classmethod
+    def properties(cls):
+        """Return the css property keys this shorthand defines."""
+        return iter(cls.VALID_KEYS)
```

Comment:
This seems like a complication that provides no benefit... why not just access VALID_KEYS directly?

Reply:
Felt cleaner 🤷‍♂ , but will remove